File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {

        List<List<Integer>> result = new ArrayList<List<Integer>>();

        levelOrder(result, root, 0);

        return result;        
    }

    public void levelOrder(List<List<Integer>> result, TreeNode root, int level) {

        if(root == null){
            return;
        }

        if(result.size() == level){
            result.add(new ArrayList<>());
        }
        result.get(level).add(root.val);      

        levelOrder(result, root.left, level+1);
        levelOrder(result, root.right, level+1);

    }
}

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:

        res = []

        def getOrderByLevel(root, level):

            #Time Complexity : O(N)
            #Space Complexity : O(N) + O(H)

            if root is None:
                return

            if len(res) == level:
                res.append([])

            res[level].append(root.val)

            getOrderByLevel(root.left, level+1)
            getOrderByLevel(root.right, level+1)            

        getOrderByLevel(root, 0)

        return res